src/services/api/base.ts

```diff
@@ -0,0 +1,52 @@
+import { request } from "https";
+
+import { ITokenProvider } from '../token/provider';
+
+export class ApiClientBase {
+
+  constructor(
+    private baseUrl: string,
+    private tokenProvider: ITokenProvider,
+  ) {
+  }
+
+  protected async request<T>(method: string, path: string, refreshToken: boolean = false): Promise<T> {
```

Comment:
```suggestion
  //type Method = "POST" | "GET" | .... 
  protected async request<T>(method: Method, path: string, refreshToken: boolean = false): Promise<T> {
```

method用のTypeを作っても良いかもしれない？

Reply:
正しい（突貫工事をゆるして）